lib/supavisor/client_handler.ex

```diff
@@ -470,7 +470,20 @@ defmodule Supavisor.ClientHandler do
     }
 
     {:ok, db_pid} = DbHandler.start_link(args)
-    db_sock = :gen_statem.call(db_pid, {:checkout, data.sock, self()})
+
+    db_sock =
+      try do
+        DbHandler.checkout(db_pid, data.sock)
+      rescue
+        err ->
+          Logger.error(
+            "ClientHandler: Error while connecting to the DB - #{Exception.message(err)}"
+          )
+
+          _ = DbHandler.stop(db_pid)
+          reraise err, __STACKTRACE__
```

Comment:
does it print creds?

Reply:
It uses "regular" error handling from Erlang, so it will use "default" view as any other `raise` that is happening in the system.

Reply:
@abc3 so in fact it changes *nothing* about the error reporting. Because without that `try` there if some function call would raise, then that error would happen in exactly the same way as it is happening with this code.